src/main/httpCommandServer.ts

```diff
@@ -118,32 +123,80 @@ export class HttpCommandServer {
     return this.dispatchTable[version]?.[method]?.[commandName];
   }
 
-  listSettings(request: http.IncomingMessage, response: http.ServerResponse) {
-    const settings = this.commandWorker?.getSettings();
+  async listSettings(request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await new Promise((resolve) => {
+      const settings = this.commandWorker?.getSettings();
 
-    if (settings) {
-      response.writeHead(200, { 'Content-Type': 'text/plain' });
-      response.write(settings);
+      if (settings) {
+        console.log('listSettings: succeeded 200');
+        response.writeHead(200, { 'Content-Type': 'text/plain' });
+        response.write(settings);
+      } else {
+        console.log('listSettings: failed 200');
+        response.writeHead(404, { 'Content-Type': 'text/plain' });
+        response.write('No settings found');
+      }
+      resolve();
+    });
+  }
+
+  /**
+   * Allow the parameters to come in both via URL parameters (non-traditional) and in a request body.
+   * @param request
+   * @param response
+   */
+  async updateSettings(request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    const url = new URL(request.url as string, `http://${ request.headers.host }`);
+    // TODO: Likely we'll be dropping support for searchParams here.
+    const searchParams = url.searchParams;
+    const chunks: Buffer[] = [];
+
+    // Read in the request body
+    for await (const chunk of request) {
+      chunks.push(chunk);
+    }
+    const data = Buffer.concat(chunks).toString();
+    // Deliberately throw a (500) exception if the data exists and isn't valid JSON
+    const values = data ? JSON.parse(data) : {};
+
+    for (const entry of searchParams.entries()) {
+      values[entry[0]] = entry[1];
+    }
+    const [result, error] = await (this.commandWorker as CommandWorkerInterface).updateSettings(values);
+
+    if (result) {
+      console.log(`updateSettings: write back status 202, result: ${ result }`);
+      response.writeHead(202, { 'Content-Type': 'text/plain' });
+      response.write(result);
     } else {
-      response.writeHead(404, { 'Content-Type': 'text/plain' });
-      response.write('No settings found');
+      console.log(`updateSettings: write back status 400, error: ${ error }`);
+      response.writeHead(400, { 'Content-Type': 'text/plain' });
+      response.write(error);
     }
   }
 
-  wrapShutdown(request: http.IncomingMessage, response: http.ServerResponse) {
-    response.writeHead(202, { 'Content-Type': 'text/plain' });
-    response.write('Shutting down.');
-    setImmediate(() => {
-      this.shutdown();
-      this.commandWorker?.requestShutdown();
+  async wrapShutdown(request: http.IncomingMessage, response: http.ServerResponse): Promise<void> {
+    return await new Promise((resolve) => {
```

Comment:
See above; no need to wrap in a `new Promise`.

Reply:
Yup.